### PR TITLE
Add documentation based on internal documentation

### PR DESCRIPTION
**Pull request by d-j-hatton**
Added some public documentation to `zocalo` based on Diamond Light Source internal documentation. Two new images are included. Added a link to this documentation from the top level `README.rst`.

### CODE REVIEW

**Comment by d-j-hatton** [comment]
@Anthchirp I have placed the new documentation in the `zocalo` directory. I don't know if this is the best place for it. It is currently written in markdown so could be moved to the top level but would require conversion to rst format

**Comment by Anthchirp** [comment]
It looks like you have changed the file formats from Unix to DOS. Could you re-unix the newlines?

**Review by Anthchirp — APPROVED** [review]

**Comment by d-j-hatton** [comment]
The documentation has been converted and moved to `README.rst`. `zocalo/README.MD` has been removed.

**Review by Anthchirp — APPROVED** [review]
Looking good. Do you want to squash-merge this?

File: README.rst [hunk 1]
```diff
@@ -43,10 +43,93 @@ Zocalo
...
+We have a special service that looks for repeat failures and moves them to a special "Dead Letter Queue". This service is called Schlockmeister_, and is the only service at time of writing that has migrated to the public zocalo repository. This service looks inside the message that got sent, extracts some basic information from the message in as safe a way as possible and repackages to the DLQ with information on what it was working on, and the "history" of where the message chain has been routed.
+
+.. _Schlockmeister: https://github.com/DiamondLightSource/python-zocalo/tree/master/zocalo/service
+
+Documentation 
+-------------
+
+Further documentation is available at https://zocalo.readthedocs.io.
 
 Features
```
Comment (Anthchirp) [inline README.rst:134]: Can remove the pointless `Features` section

Comment (d-j-hatton) [reply README.rst]: Done. I'll merge once the checks have passed

**Pull request merged**